### PR TITLE
Pharmacy it v3

### PR DESCRIPTION
**Pull request by fansanelli**
I'm sorry for the noise. I hope these classes are correct and I also removed the "missing_osm" because datasets exclude each other. Please also consider deleting all 7150 errors in production.

### CODE REVIEW

**Comment by frodrigo** [comment]
Adding `dispensing` in the select (only for parapharmacy ?) should do the job for missing_osm, I think.

**Comment by fansanelli** [comment]
Sure. I could hack around it during the weekend... Can we go ahead with this in the meanwhile?

**Comment by frodrigo** [comment]
Ok. But so keep the `missing_osm` we will disable it in the frontend.

**Comment by fansanelli** [comment]
Ok. I read how analyzer works and hope to have fixed the missing_osm.
This is untested... please let me know if classes are fine.
Thanks

File: analysers/analyser_merge_parapharmacy_IT.py [hunk 1]
```diff
@@ -27,13 +27,13 @@
 class Analyser_Merge_Parapharmacy_IT(Analyser_Merge):
     def __init__(self, config, logger = None):
         Analyser_Merge.__init__(self, config, logger)
-        self.missing_official = self.def_class(item = 7150, id = 21, level = 3, tags = ['merge', 'highway'],
+        self.missing_official = self.def_class(item = 8210, id = 21, level = 3, tags = ['merge', 'highway'],
             title = T_('Pharmacy not integrated'))
-        self.missing_osm      = self.def_class(item = 7250, id = 22, level = 3, tags = ['merge', 'highway'],
+        self.missing_osm      = self.def_class(item = 8212, id = 22, level = 3, tags = ['merge', 'highway'],
```
Comment (frodrigo) [inline analysers/analyser_merge_parapharmacy_IT.py]: No, `missing_osm` should not chnage and stay at `7250`

**Comment by frodrigo** [comment]
Ok. Fix the `missing_osm` and I will test it.

File: analysers/analyser_merge_parapharmacy_IT.py (hunk 1 above)
Comment (fansanelli) [reply analysers/analyser_merge_parapharmacy_IT.py]: done

File: analysers/analyser_merge_parapharmacy_IT.py [hunk 2]
```diff
@@ -27,13 +27,13 @@
...
-        self.missing_official = self.def_class(item = 7150, id = 21, level = 3, tags = ['merge', 'highway'],
+        self.missing_official = self.def_class(item = 8210, id = 21, level = 3, tags = ['merge', 'highway'],
             title = T_('Pharmacy not integrated'))
         self.missing_osm      = self.def_class(item = 7250, id = 22, level = 3, tags = ['merge', 'highway'],
             title = T_('Pharmacy without tag `ref:msal` or invalid'))
-        self.possible_merge   = self.def_class(item = 7151, id = 23, level = 3, tags = ['merge', 'highway'],
+        self.possible_merge   = self.def_class(item = 8211, id = 23, level = 3, tags = ['merge', 'highway'],
             title = T_('Pharmacy integration suggestion'))
-        self.update_official  = self.def_class(item = 7152, id = 24, level = 3, tags = ['merge', 'highway'],
+        self.update_official  = self.def_class(item = 8211, id = 24, level = 3, tags = ['merge', 'highway'],
```
Comment (frodrigo) [inline analysers/analyser_merge_parapharmacy_IT.py]:
```suggestion
        self.update_official  = self.def_class(item = 8212, id = 24, level = 3, tags = ['merge', 'highway'],
```

File: analysers/analyser_merge_pharmacy_IT.py [hunk 3]
```diff
@@ -27,13 +27,13 @@
...
-        self.missing_official = self.def_class(item = 7150, id = 11, level = 3, tags = ['merge', 'highway'],
+        self.missing_official = self.def_class(item = 8210, id = 11, level = 3, tags = ['merge', 'highway'],
             title = T_('Pharmacy not integrated'))
         self.missing_osm      = self.def_class(item = 7250, id = 12, level = 3, tags = ['merge', 'highway'],
             title = T_('Pharmacy without tag `ref:msal` or invalid'))
-        self.possible_merge   = self.def_class(item = 7151, id = 13, level = 3, tags = ['merge', 'highway'],
+        self.possible_merge   = self.def_class(item = 8211, id = 13, level = 3, tags = ['merge', 'highway'],
             title = T_('Pharmacy integration suggestion'))
-        self.update_official  = self.def_class(item = 7152, id = 14, level = 3, tags = ['merge', 'highway'],
+        self.update_official  = self.def_class(item = 8211, id = 14, level = 3, tags = ['merge', 'highway'],
```
Comment (frodrigo) [inline analysers/analyser_merge_pharmacy_IT.py]:
```suggestion
        self.update_official  = self.def_class(item = 8212, id = 24, level = 3, tags = ['merge', 'highway'],
```

File: analysers/analyser_merge_pharmacy_IT.py [hunk 4]
```diff
@@ -33,7 +33,7 @@ def __init__(self, config, logger = None):
             title = T_('Pharmacy without tag `ref:msal` or invalid'))
         self.possible_merge   = self.def_class(item = 8211, id = 13, level = 3, tags = ['merge', 'highway'],
             title = T_('Pharmacy integration suggestion'))
-        self.update_official  = self.def_class(item = 8211, id = 14, level = 3, tags = ['merge', 'highway'],
+        self.update_official  = self.def_class(item = 8212, id = 24, level = 3, tags = ['merge', 'highway'],
```
Comment (fansanelli) [inline analysers/analyser_merge_pharmacy_IT.py]: id 24 or 14?

Comment (frodrigo) [reply analysers/analyser_merge_pharmacy_IT.py]: 14

**Comment by frodrigo** [comment]
After review what `extraJoin` is, I think it is a complicate way to the same as adding `dispensing` in the select since `osm_item.tags->'%(tag)s'` is always the same value by analysis: `dispensing=yes|no`.

**Comment by fansanelli** [comment]
I'm not sure to be able of doing this right now... I'll check again tomorrow

**Comment by fansanelli** [comment]
If I read the code well, changing the selection will miss the opportunity to merge pharmacy without tag dispensing, while extraJoin apply only to missing osm. I probably may have missed something...

**Comment by frodrigo** [comment]
No. It is also in use by others.

**Comment by fansanelli** [comment]
Hello,
I hope now it's fine... I haven't tested.

**Comment by frodrigo** [comment]
Thank merged. Not deployed yet.

**Comment by fansanelli** [comment]
Thanks! I'm sorry it took so long

**Comment by frodrigo** [comment]
Don't worry about that.
Have you try to use the docker to test your code ? https://github.com/osm-fr/osmose-backend/tree/master/docker

**Comment by fansanelli** [comment]
I'm using it ... But haven't hook a frontend yet... I this case it would have been useful

**Pull request merged**